### PR TITLE
Add DB cache for sessions

### LINKED ISSUE #156: DB cache for sessions
**Idea Description**
The timetable loading is sometimes long 😭 
We should cache it. Can we use sqldelight?

**Reference images and links**
https://github.com/cashapp/sqldelight

### PR DESCRIPTION
**Pull request by fornewid**
## Issue
- close #156 

## Overview (Required)
- Install SQLDelight
- Add `"droidkaigi.primitive.sqldelight"` plugin into convention plugins.
- Add `SessionDao`, `DatabaseService`, `DriverFactory` for database
   and prepare DI modules (Android: `DaoModule`, iOS: `KoinModules`).
- Add DB tables for `TimetableItem.Session`, `TimetableItem.Special`, `TimetableSpeaker`.
- In `DataSessionsRepository`, adjust logics about both `droidKaigiScheduleFlow()` and `refresh()` using `SessionDao`.

## Links
- https://cashapp.github.io/sqldelight/multiplatform_sqlite/
- https://github.com/DroidKaigi/conference-app-2021/pull/275

### CODE REVIEW

**Comment by fornewid** [comment]
error logs:
```
e: /home/runner/work/conference-app-2022/conference-app-2022/core-data/src/
  commonMain/kotlin/io/github/droidkaigi/confsched2022/data/DriverFactory.kt: (5, 14):
  Expected class 'DriverFactory' has no actual declaration in module <core-data_debug> for JVM

e: /home/runner/work/conference-app-[202](https://github.com/DroidKaigi/conference-app-2022/runs/8244330205?check_suite_focus=true#step:11:203)2/conference-app-2022/core-data/src/
  commonMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/SessionsDao.kt: (174, 35):
  Smart cast to 'String' is impossible, because 'session.messageJa' is a public API property declared in different module
e: /home/runner/work/conference-app-2022/conference-app-2022/core-data/src/
  commonMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/SessionsDao.kt: (175, 35):
  Smart cast to 'String' is impossible, because 'session.messageEn' is a public API property declared in different module
```

**Comment by takahirom** [comment]
👀

**Comment by takahirom** [comment]
I experienced a similar error, and It took me one hour to remind it..
What you have to do is create androidTest folder and implement the class.
Please see core-model/androidTest folder.

**Comment by takahirom** [comment]
> class NativeDriverFactory should be declared in a file named NativeDriverFactory.kt
java.lang.AssertionError: Error on line: 1, column: 1

Please check it and  run ./gradlew spotlessKotlinApply 🙏

**Comment by fornewid** [comment]
I fixed CI errors. :bow:

File: core-data/src/commonMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/DataSessionsRepository.kt [hunk 1]
```diff
@@ -3,25 +3,22 @@ package io.github.droidkaigi.confsched2022.data.sessions
...
 ) : SessionsRepository {
-    private val timetableFlow = MutableStateFlow(Timetable())
     override fun droidKaigiScheduleFlow(): Flow<DroidKaigiSchedule> = callbackFlow {
-        if (timetableFlow.value.isEmpty()) {
-            refresh()
-        }
+        launch { refresh() }
         combine(
-            timetableFlow,
+            sessionsDao.selectAll(),
```
Comment (takahirom) [inline core-data/src/commonMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/DataSessionsRepository.kt:21]: Great SSoT👍

**Review by takahirom — APPROVED** [review]
Thank you for creating DroidKaigi app's Single source of truth!

File: core-data/src/commonMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/DataSessionsRepository.kt (hunk 1 above)
Comment (fornewid) [reply core-data/src/commonMain/kotlin/io/github/droidkaigi/confsched2022/data/sessions/DataSessionsRepository.kt:21]: Thanks! 😄

**Pull request merged**